### PR TITLE
[cicd] trigger forge from GHA

### PR DESCRIPTION
**Pull request by geekflyer**
this moves the forge triggering from circleci to github actions.

### CODE REVIEW

File: .github/workflows/run-forge.yaml [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+          FORGE_REPORT_TXT=$(cat forge_report.json | jq -r .text)
+          echo "FORGE_REPORT_TXT='$FORGE_REPORT_TXT'" >> $GITHUB_ENV
+
+          echo "Forge run: $GITHUB_SERVER_URL/$GITHUB_REPOSITORY/actions/runs/$GITHUB_RUN_ID" >> forge_comment.txt
+          echo "Forge test result: $FORGE_REPORT_TXT" >> forge_comment.txt
+
+          # replace all newlines
+          FORGE_COMMENT=$(awk '{printf "%s\\n", $0}' forge_comment.txt)
+
+          # post  comment on the PR
```
Comment (rustielin) [inline .github/workflows/run-forge.yaml:65]: now that we're using GHA, can we do this in an action step? maybe https://github.com/marketplace/actions/add-pr-comment

File: .github/workflows/build-images.yaml [hunk 2]
```diff
@@ -59,12 +71,23 @@ jobs:
     needs: rust-images
     uses: ./.github/workflows/sdk-integration-test.yaml
     secrets: inherit
+    with:
+      GIT_SHA: ${{ github.event.pull_request.head.sha || github.sha }}
```
Comment (rustielin) [inline .github/workflows/build-images.yaml:75]: is it possible to do:

```
with:
  GIT_SHA: ${{ .env.GIT_SHA }}
```

so we can reuse the env defined at the top of the workflow?

Comment (geekflyer) [reply .github/workflows/build-images.yaml:75]: just tried it out. unfortunately we can't - https://github.com/aptos-labs/aptos-core/actions/runs/2450658591 . Somehow `env` is not available in the context of calling another workflow.

File: .github/workflows/run-forge.yaml (hunk 1 above)
Comment (geekflyer) [reply .github/workflows/run-forge.yaml]: done

**Review by rustielin — APPROVED** [review]

**Comment by rustielin** [comment]
This PR LGTM, but looks like Forge is broken now. From the last workflow:

```
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: failed to helm get values aptos: failed to deserialize helm values: EOF while parsing a value at line 1 column 0
```

Try resetting the cluster?

**Pull request closed without merging**